src/pause/138CC0.c

```diff
@@ -103,19 +103,386 @@ s32 pause_badges_scroll_offset_x(s32 x) {
     return x;
 }
 
-INCLUDE_ASM(s32, "pause/138CC0", pause_badges_try_remove);
+s32 pause_badges_try_remove(s16 badgeID) {
+    s16 *currentSlot = gPlayerData.equippedBadges;
+    s16 *slotToRemove = currentSlot;
+    s32 result = 0;
+    s32 i;
+
+    if (badgeID == 0x7FFF) {
+        return 0;
+    }
+
+    for (i = 0; i < ARRAY_COUNT(gPlayerData.equippedBadges); ++i, ++currentSlot) {
+        if (badgeID == *currentSlot) {
+            slotToRemove = currentSlot;
+            result = 1;
+            break;
+        }
+    }
+
+    // Remove and shift array contents
+    for (; i < ARRAY_COUNT(gPlayerData.equippedBadges)-1; ++i, ++currentSlot) {
+        *currentSlot = *(currentSlot+1);
```

Comment:
you should be able to do *currentSlot++ (parens may be necessary around *currentSlot - I can't remember here)

Reply:
`*currentSlot++` would not be well formed here, iirc in C there's no sequence point between expressions on the left and right side of an equals sign. It wouldn't be well defined whether the RHS uses the incremented or not incremented version of `currentSlot`.

Regardless, the increment being in the for loop header does actually matter, it changes where the increment operation gets emitted in the asm (I had to massage this loop a bunch to get it right)